src/binding/axum/extract.rs

```diff
@@ -0,0 +1,143 @@
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        let headers = req
+            .headers()
+            .cloned()
+            .ok_or(0)
+            .map_err(|_| (StatusCode::BAD_REQUEST, "Invalid Cloud Event"))?;
+
+        let req_body = req
+            .take_body()
+            .ok_or(0)
+            .map_err(|_| (StatusCode::BAD_REQUEST, "Invalid Cloud Event"))?;
+
+        let buf = body::to_bytes(req_body)
+            .await
+            .map_err(|_| (StatusCode::BAD_REQUEST, ""))?;
+        to_event(&headers, buf.to_vec())
+            .map_err(|_| (StatusCode::BAD_REQUEST, "Invalid Cloud Event"))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use axum_lib as axum;
+
+    use super::*;
+    use axum::body::Body;
+    use axum::http::{self, Request, StatusCode};
+    use chrono::Utc;
+    use serde_json::json;
+
+    use crate::{EventBuilder, EventBuilderV10};
+
+    #[tokio::test]
+    async fn axum_test_request() {
+        let time = Utc::now();
+        let expected = EventBuilderV10::new()
+            .id("0001")
+            .ty("example.test")
+            .source("http://localhost/")
+            .time(time)
+            .extension("someint", "10")
+            .build()
+            .unwrap();
+
+        let mut request = RequestParts::new(
+            Request::builder()
+                .method(http::Method::POST)
+                .header("ce-specversion", "1.0")
+                .header("ce-id", "0001")
+                .header("ce-type", "example.test")
+                .header("ce-source", "http://localhost/")
+                .header("ce-someint", "10")
+                .header("ce-time", time.to_rfc3339())
+                .body(Body::empty())
+                .unwrap(),
+        );
+
+        let result = Event::from_request(&mut request).await.unwrap();
+
+        assert_eq!(expected, result);
+    }
+
+    #[tokio::test]
+    async fn axum_test_bad_request() {
+        let time = Utc::now();
+
+        let mut request = RequestParts::new(
+            Request::builder()
+                .method(http::Method::POST)
+                .header("ce-specversion", "BAD SPECIFICATION")
+                .header("ce-id", "0001")
+                .header("ce-type", "example.test")
+                .header("ce-source", "http://localhost/")
+                .header("ce-someint", "10")
+                .header("ce-time", time.to_rfc3339())
+                .body(Body::empty())
+                .unwrap(),
+        );
+
+        let result = Event::from_request(&mut request).await;
+        assert!(result.is_err());
+        let rejection = result.unwrap_err();
+
+        let reason = rejection.0;
+        assert_eq!(reason, StatusCode::BAD_REQUEST)
+    }
+
+    #[tokio::test]
+    async fn axum_test_request_with_full_data() {
+        let time = Utc::now();
+        let j = json!({"hello": "world"});
+
+        let expected = EventBuilderV10::new()
```

Comment:
I just notices that we have still work to do on using actual fixtures in unit tests. I'll jump on this next and then we can change these as well.

Reply:
Thank you for your feedback.
Would love to learn what that suggestion looks like in your pull request so I can support by example.